test/compile/data/dataflow.test.ts

```diff
@@ -70,15 +70,41 @@ describe('compile/data/dataflow', () => {
 
         const c = new DataFlowNode(b, 'c');
 
-        assert.deepEqual(a.children, [b]);
+        expect(a.children).toEqual([b]);
         assert.equal(b.parent, a);
         assert.equal(c.parent, b);
 
         b.remove();
 
-        assert.deepEqual(a.children, [c]);
+        expect(a.children).toEqual([c]);
         assert.equal(c.parent, a);
       });
+
+      it('should maintain order', () => {
+        const root = new DataFlowNode(null, 'root');
+
+        const rootChild1 = new DataFlowNode(root, 'rootChild1');
+        const node = new DataFlowNode(root, 'node');
+        const rootChild2 = new DataFlowNode(root, 'rootChild2');
+
+        const child1 = new DataFlowNode(node, 'child1');
+        const child2 = new DataFlowNode(node, 'child2');
+
+        expect(root.children).toEqual([rootChild1, node, rootChild2]);
+        assert.equal(rootChild1.parent, root);
```

Comment:
No need to change for this PR, but let's always use `expect` in new code. 
So it will be less work to remove chai later. 

Note that the auto-migrating tool doesn't work for TS yet. TT

Reply:
I changed it because I wanted to get a good diff for this PR :-P

Reply:
I mean you add `assert.equal` below, which could have been `expect(...).toEqual()`. :p

Reply:
Oh yeah, I copied a test.